libyul/Object.cpp

```diff
@@ -65,9 +65,45 @@ set<YulString> Object::dataNames() const
 {
 	set<YulString> names;
 	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
-	// The empty name is not valid
+	for (shared_ptr<ObjectNode> const& subObjectNode: subObjects)
+	{
+		names.insert(subObjectNode->name);
+		if (auto const* subObject = dynamic_cast<Object const*>(subObjectNode.get()))
+			for (YulString const& subSubObj: subObject->dataNames())
+				names.insert(YulString{subObject->name.str() + "." + subSubObj.str()});
```

Comment:
https://github.com/ethereum/solidity/pull/9047#discussion_r431783054

```suggestion
			for (YulString const& qualifiedSubSubObjectName: subObject->dataNames())
				names.insert(YulString{subObject->name.str() + "." + qualifiedSubSubObjectName.str()});
```